modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -115,20 +116,20 @@ object EnrichmentManager {
     registryLookup: RegistryLookup[F]
   ): EitherT[F, BadRow, IgluUtils.EventExtractResult] =
     EitherT {
-      val setup = setupEnrichedEvent(raw, enrichedEvent, etlTstamp, processor).toValidatedNel
-      IgluUtils
-        .extractAndValidateInputJsons(enrichedEvent, client, registryLookup)
-        .map((setup, _).mapN((_, extract) => extract))
-        .map {
-          _.leftMap { violations =>
-            buildSchemaViolationsBadRow(
-              violations,
-              EnrichedEvent.toPartiallyEnrichedEvent(enrichedEvent),
-              RawEvent.toRawEvent(raw),
-              processor
-            )
-          }.toEither
+      for {
+        setup <- setupEnrichedEvent[F](raw, enrichedEvent, etlTstamp, processor).map(_.toValidatedNel)
+        iglu <- IgluUtils.extractAndValidateInputJsons(enrichedEvent, client, registryLookup)
+      } yield (setup, iglu)
+        .mapN((_, extract) => extract)
+        .leftMap { violations =>
```

Comment:
How about

```scala
} yield (iglu <* setup)
  .leftMap { violations =>
```

or 

```scala
} yield iglu
  .productL(setup)
  .leftMap { violations =>
```

Personally I find those easier to understand than `.mapN((_, extract) => extract)` but it's fine if you disagree!

Reply:
Oh yeah you had already told me and I forgot, sorry ! Agree that it looks better